microdnf/commands/repo/repo_list.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "repo_list.hpp"
+
+#include "../../context.hpp"
+
+#include <libdnf/conf/option_string.hpp>
+#include "libdnf-cli/output/repolist.hpp"
+
+
+namespace microdnf {
+
+
+using namespace libdnf::cli;
+
+
+RepoListCommand::RepoListCommand(Command & parent) : RepoListCommand(parent, "list") {}
+
+
+RepoListCommand::RepoListCommand(Command & parent, const std::string & name) : Command(parent, name) {
+    auto & ctx = static_cast<Context &>(get_session());
+    auto & parser = ctx.get_argument_parser();
+
+    auto & cmd = *get_argument_parser_command();
+    cmd.set_short_description("List defined repositories");
+
+    all = std::make_unique<RepoAllOption>(*this);
+    enabled = std::make_unique<RepoEnabledOption>(*this);
+    disabled = std::make_unique<RepoDisabledOption>(*this);
+    repo_specs = std::make_unique<RepoSpecArguments>(*this);
+
+    auto conflict_args =
+        parser.add_conflict_args_group(std::unique_ptr<std::vector<ArgumentParser::Argument *>>(
+            new std::vector<ArgumentParser::Argument *>{all->arg, enabled->arg, disabled->arg}));
+
+    all->arg->set_conflict_arguments(conflict_args);
+    enabled->arg->set_conflict_arguments(conflict_args);
+    disabled->arg->set_conflict_arguments(conflict_args);
+
+    /*
+    auto repoinfo = parser.add_new_command("repoinfo");
+    repoinfo->set_short_description("Print detais about defined repositories");
+    repoinfo->set_parse_hook_func([this, &ctx](
+                               [[maybe_unused]] ArgumentParser::Argument * arg,
+                               [[maybe_unused]] const char * option,
+                               [[maybe_unused]] int argc,
+                               [[maybe_unused]] const char * const argv[]) {
+        // TODO(jrohel): implement repoinfo
+        throw std::logic_error("Not implemented");
+        ctx.set_selected_command(this);
+        return true;
+    });
+
+    repoinfo->register_named_arg(all);
+    repoinfo->register_named_arg(enabled);
+    repoinfo->register_named_arg(disabled);
+    repoinfo->register_positional_arg(repos);
+
+    parser.get_root_command()->register_command(repoinfo);
+    */
```

Comment:
You implemented the `repo info` command and alias `repoinfo` in the last commit. Remove comment out code above.

Reply:
Fixed